pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +98,248 @@ func (builder *Builder) validate() (bool, error) {
+	var err error
+
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	if desiredUpdateImage == "" {
+		glog.V(100).Infof("The desiredUpdateImage is empty")
+
+		return builder, fmt.Errorf("'desiredUpdateImage' cannot be empty")
+	}
+
+	glog.V(100).Info("Updating clusterversion %s to the desired image %s",
+		builder.Definition.Name, desiredUpdateImage)
+
+	builder.Definition.Spec.DesiredUpdate = &v1.Update{Image: desiredUpdateImage, Force: force}
+
+	return builder, err
+}
+
+// WithDesiredUpdateChannel adds the desired channel to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateChannel(updateChannel string) (*Builder, error) {
+	var err error
+
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	if updateChannel == "" {
+		glog.V(100).Infof("The updateChannel is empty")
+
+		return builder, fmt.Errorf("'updateChannel' cannot be empty")
+	}
+
+	glog.V(100).Info("Updating clusterversion %s to the desired updateChannel %s",
+		builder.Definition.Name, updateChannel)
+
+	builder.Definition.Spec.Channel = updateChannel
+
+	return builder, err
+}
+
+// Update renovates the existing clusterversion object with the clusterversion definition in builder.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating configuration of clusterversion")
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("clusterversion object does not exist")
+	}
+
+	builder.Definition.CreationTimestamp = metav1.Time{}
+
+	var err error
+	builder.Object, err = builder.apiClient.ConfigV1Interface.ClusterVersions().Update(
+		context.TODO(), builder.Definition, metav1.UpdateOptions{})
+
+	return builder, err
+}
+
+// Get fetches existing clusterversion from cluster.
+func (builder *Builder) Get() (*v1.ClusterVersion, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting existing clusterversion with name %s from cluster", builder.Definition.Name)
+
+	clusterVersionObj := &v1.ClusterVersion{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, clusterVersionObj)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get clusterversion object %s from cluster due to: %w",
+			builder.Definition.Name, err)
+
+		return nil, err
+	}
+
+	return clusterVersionObj, nil
+}
+
+// WaitUntilProgressing waits for timeout duration or until clusterversion is in Progressing state.
+func (builder *Builder) WaitUntilProgressing(timeout time.Duration) error {
+	return builder.WaitUntilConditionTrue("Progressing", timeout)
+}
+
+// WaitUntilAvailable waits for timeout duration or until clusterversion is in Available state.
+func (builder *Builder) WaitUntilAvailable(timeout time.Duration) error {
+	return builder.WaitUntilConditionTrue("Progressing", timeout)
```

Comment:
I think we have wrong conditionType here Progressing instead of Available

Reply:
Changed, thanks